tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,793 @@
+            'Possibly_noisy': p_noisy,
+            'Slice_range': slice_range,
+            'Spacing_mm_px_': spacing,
+            'DICOM_windows': dicom_windows,
+            'Patient_gender': p_gender,
+            'Patient_age': p_age,
+            'Train_Val_Test': sp,
+        }
+        normal_scans_ann.append(normal)
+
+    # create a new Dataframe of normals
+    pd = tfds.core.lazy_imports.pandas
+    normal_ann = pd.DataFrame(
+        normal_scans_ann,
+        columns=[
+            'File_name', 'Bounding_boxes', 'Image_size', 'Key_slice_index',
+            'Measurement_coordinates', 'Lesion_diameters_Pixel_',
+            'Normalized_lesion_location', 'Possibly_noisy', 'Slice_range',
+            'Spacing_mm_px_', 'DICOM_windows', 'Patient_gender',
+            'Patient_age', 'Train_Val_Test',
+        ])
+    normal_ann = normal_ann.drop_duplicates("File_name")
+    mask = normal_ann["File_name"].isin(ann["File_name"])
+    normal_ann = normal_ann[~mask]
+
+    return normal_ann.sample(n=len(ann), random_state=42)
+
+
+def _cal_cutoff_position(length, thickness, keyIdx):
+  """Returns cutoff indices of keyIdx-centred truncated list
+
+  Args:
+    length: `int`, length of a list
+    thickness: `int`, cutoff thickness - number of slices
+    keyIdx: `int`, index of the key slice
+
+  Returns:
+    a tuple of two `int`
+  """
+  left_block = (thickness - 1) // 2
+  right_block = thickness - 1 - left_block
+
+  start = max(0, keyIdx - left_block)
+  end = min(length - 1, keyIdx + right_block)
+  return start, end
+
+
+def _format_bboxs(bboxs, size):
+  """Returns bbox feature
+
+  Args:
+    bboxs: `str`, "xmin,ymin,xmax,ymax"
+    size: `str`, "height,width", height is assumed to be equal with width.
+
+  Returns:
+    tfds.features.BBox
+  """
+  size = [float(x) for x in size.split(',')]
+  if len(size) != 2 or size[0] != size[1]:
+    raise AssertionError(
+        'height should be equal with width for this dataset')
+  coords = np.clip([float(x) for x in bboxs.split(',')], 0.0, size[0])
+
+  cnt = int((len(coords)) / 4)
+
+  bbox_list = []
+  for i in range(cnt):
+    ymin = coords[i * 4 + 1] / size[1]
+    xmin = coords[i * 4] / size[0]
+    ymax = coords[i * 4 + 3] / size[1]
+    xmax = coords[i * 4 + 2] / size[0]
+    bbox_list.append(
+        tfds.features.BBox(ymin=ymin, xmin=xmin, ymax=ymax, xmax=xmax))
+
+  return bbox_list
+
+
+def _format_values(val, n, e_type):
+  """Returns a feature formated value
+
+  Args:
+    val: `str`, "num1,num2,num3,...", value to format
+    n: `int`, number of elememts within tuples
+    e_type: `str`, the type of each elements within tuples
+
+  Returns
+    a list of tuples of e_type values
+  """
+  def _divide_chunks(l, n):
+    """Yield successive n-sized chunks from l.
+    """
+    for i in range(0, len(l), n):
+      yield tuple(l[i:i + n])
+
+  lst = [e_type(float(x)) for x in val.split(',')]
+  return list(_divide_chunks(lst, n))
```

Comment:
if you're returning a list anyway, you probably don't need a generator.

return [ lst[i:i+n] for i in range(0, len(lst), n)]